dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESMappingAPITest.java

```diff
@@ -572,4 +573,83 @@ public void testLoadCategories_GivenContentWithCats_ShouldLoadESMapWithListOfCat
+     * Expected Results: Then we feed data into such field and use an ES query over the KeyValue to verify the results are coming back.
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @Test
+    public void Test_Create_ContentType_With_KeyValue_Field_Test_Query_Expect_Success()
+            throws DotDataException, DotSecurityException {
+
+        List<Field> fields = new ArrayList<>();
+        final String myKeyValueField = "myKeyValueField";
+        fields.add(
+                new FieldDataGen()
+                        .type(KeyValueField.class)
+                        .name(myKeyValueField)
+                        .velocityVarName(myKeyValueField).indexed(true)
+                        .next()
+        );
+        final String contentTypeName = "myContentTypeWithKeyValues" + System.currentTimeMillis();
+        final ContentType contentType = new ContentTypeDataGen()
+                .name(contentTypeName)
+                .velocityVarName(contentTypeName)
+                .fields(fields)
+                .nextPersisted();
+
+        new ContentletDataGen(contentType.id()).setProperty(myKeyValueField,
+                "{\"key\":\"key1\", value:\"val\" }").nextPersisted();
+        final String queryString =  String.format("+%s.%s.key:%s*",contentTypeName, myKeyValueField, "val");
+        Logger.info(ESMappingAPITest.class, () -> String.format(" Query: %s ",queryString));
+        final String wrappedQuery = String.format("{"
+                + "query: {"
+                + "   query_string: {"
+                + "        query: \"%s\""
+                + "     }"
+                + "  }"
+                + "}", queryString);
+
+        final ESSearchResults searchResults = contentletAPI.esSearch(wrappedQuery, false,  user, false);
+        Assert.assertFalse(searchResults.isEmpty());
+        for (Object searchResult : searchResults) {
+           final Contentlet contentlet = (Contentlet) searchResult;
+            final String json = (String)contentlet.getMap().get("myKeyValueField");
+            Assert.assertNotNull(json);
+            final Map<String, Object> map = KeyValueFieldUtil.JSONValueToHashMap(json);
+            Assert.assertTrue(map.containsKey("key"));
+        }
+    }
+
+    /**
+     * General purpose is testing that Metadata fields are indexed correctly can be used as query parameters.
+     * Given scenario: Known that saving a file assets generates metadata into the index. We create a file asset.
+     * Expected Results: Once the file is saved we use an ES query to verify that queries by metadata are functional.
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @Test
+    public void Test_Create_FileAsset_Query_by_MetaData_Expect_Success()
+            throws DotDataException, DotSecurityException {
+
+        final File binary = new File(ESMappingAPITest.class.getClassLoader().getResource("images/test.jpg").getFile());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [In J2EE, getClassLoader() might not work as expected.  Use Thread.currentThread().getContextClassLoader() instead.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5533307)